src/Symfony/Component/Security/Core/Signature/SignatureHasher.php

```diff
@@ -90,9 +115,11 @@ public function computeSignatureHash(UserInterface $user, int $expires): string
             if (!\is_scalar($value) && !(\is_object($value) && method_exists($value, '__toString'))) {
                 throw new \InvalidArgumentException(sprintf('The property path "%s" on the user object "%s" must return a value that can be cast to a string, but "%s" was returned.', $property, \get_class($user), get_debug_type($value)));
             }
-            $signatureFields[] = base64_encode($value);
+            hash_update($fieldsHash, ':'.base64_encode($value));
         }
 
-        return base64_encode(hash_hmac('sha256', implode(':', $signatureFields), $this->secret));
+        $fieldsHash = strtr(base64_encode(hash_final($fieldsHash, true)), '+/=', '-_~');
```

Comment:
is it valid to call `hash_final` without any `hash_update` call ? This happens in your PR is `signatureProperties` is empty (which is a valid use case)

Reply:
yep, no pb: https://3v4l.org/GJ5E5